### PR TITLE
Support user-defined onStart behavior

### PR DESCRIPTION
**Pull request by tgross**
This PR provides https://github.com/joyent/containerbuddy/issues/13

- Runs a user-defined `onStart` executable prior to running the main application or any of the health check or discovery goroutines.
- Made config parsing more testable and expanded test coverage

@misterbisson to review please

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +82,10 @@ Backend fields:
 - `poll` is the time in seconds between polling for changes.
 - `onChange` is the executable (and its arguments) that is called when there is a change in the list of IPs and ports for this backend.
 
+Other fields:
+- `consul` is the hostname and port of the Consul discovery service.
+- `onStart` is the executable (and its arguments) that will be called immediately prior to starting the shimmed application. This field is optional.
```
Comment (misterbisson) [inline README.md]: Does the exit code for the `onStart` matter?

Comment (tgross) [inline README.md]: It does -- we exit if `onStart` fails. I'll add a note to that effect here.

**Comment by tgross** [comment]
I've added a `cover` target to the makefile that runs the go coverage tool with an HTML heatmap output. This shows there's a lack of test coverage of `main.go` that could use some work. I'll hit this as I address additional features.

**Comment by misterbisson** [comment]
:house_with_garden: :walking:

**Comment by tgross** [comment]
Included in release: https://github.com/joyent/containerbuddy/releases/tag/0.0.2-alpha

**Pull request merged**